From a4c35b648876c488770c07dfd2170d0da507d35b Mon Sep 17 00:00:00 2001
From: Tony LIU <junjie.liu@freescale.com>
Date: Mon, 18 Jun 2012 13:36:59 +0800
Subject: [PATCH 324/449] ENGR00213929 Add two new test case and remove two
 out of date one

- add two new test case
  stress test including gadget driver built in
  3G modem simple test case
- remove two test case
  USB host PTP case
  USB UVC test

Signed-off-by: Tony LIU <junjie.liu@freescale.com>
---
 test/doc/ut_um.toc                       |  4 +--
 test/mxc_usb_test/doc/FSL-UT-USB-016.txt | 47 +++++++++++++++++++++++++++
 test/mxc_usb_test/doc/FSL-UT-USB-017.txt | 56 ++++++++++++++++++++++++++++++++
 3 files changed, 105 insertions(+), 2 deletions(-)
 create mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-016.txt
 create mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-017.txt

diff --git a/test/doc/ut_um.toc b/test/doc/ut_um.toc
index efc0971..bc78083 100644
--- a/test/doc/ut_um.toc
+++ b/test/doc/ut_um.toc
@@ -129,19 +129,19 @@ USB
 ---
 include::../mxc_usb_test/doc/FSL-UT-USB-001.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-002.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-003.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-004.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-005.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-006.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-007.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-008.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-009.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-010.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-011.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-012.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-013.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-014.txt[]
 include::../mxc_usb_test/doc/FSL-UT-USB-015.txt[]
+include::../mxc_usb_test/doc/FSL-UT-USB-016.txt[]
+include::../mxc_usb_test/doc/FSL-UT-USB-017.txt[]
 
 V4L2
 ----
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-016.txt b/test/mxc_usb_test/doc/FSL-UT-USB-016.txt
new file mode 100644
index 0000000..fafef77
--- /dev/null
+++ b/test/mxc_usb_test/doc/FSL-UT-USB-016.txt
@@ -0,0 +1,47 @@
+FSL-UT-USB-016-Suspend-Resume-Stress-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+USB suspend and resume stress test
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. USB gadget is built in kernel
+2. a hub attached to Host1,
+3. a USB mouse attached to the hub which attached to Host1, and enable auto suspend by
+   echo auto > /sys/devices/platform/fsl-ehci.1/usb2/2-1/2-1.x/power/control
+4. another device(any other board running g_zero gadget driver) attached to OTG
+5. enable remote wake up feature for the device attached to OTG by
+   echo enabled > /sys/devices/platform/fsl-ehci.0/power/wakeup
+   echo enabled > /sys/devices/platform/fsl-usb2-otg/power/wakeup
+   echo enabled > /sys/devices/platform/fsl-ehci.0/usb1/1-1/power/wakeup
+6. no the device side which attached to OTG, set the start, end, interval parameter
+
+7. use the attached script to run suspend/resume overnight
+
+#!/bin/sh
+# This will only run the quickhit tests.
+# i=0; while [ "$i" -lt 20000 ]; do /unit_tests/rtcwakeup.out -d rtc0 -m mem -s 2; i=`expr $i + 1`; done
+i=0; t=50; d=5; r=0; while [ "$i" -lt 200000 ]; do echo mem > /sys/power/state; sleep 2;
+echo ******suspend $i times**********; i=`expr $i + 1`; r=`date +%s`; t=`expr $r % 10 + 50`; d=`expr $r % 30 + 1`; done
+
+| Expected Result |
+USB can work normally after resume
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-017.txt b/test/mxc_usb_test/doc/FSL-UT-USB-017.txt
new file mode 100644
index 0000000..45a9ac2
--- /dev/null
+++ b/test/mxc_usb_test/doc/FSL-UT-USB-017.txt
@@ -0,0 +1,56 @@
+FSL-UT-USB-017-3G-Modem-Basic-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+USB 3G modem basic test
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. make menuconfig
+   select "USB serial Converter support"
+   and the "USB Generic Serial Driver" & "USB driver for GSM and CDMA modems" under item
+   "USB serial Converter support"
+2. make uImage
+3. board power up and load kernel
+4. verify ttyUSB* exist under /dev
+5. run the following to receive 3G modem's response
+   cat /dev/ttyUSB0 &
+   cat /dev/ttyUSB1 &
+6. send AT command to 3G modem
+   echo "ATI^M" > /dev/ttyUSB1
+7. some response will be received like below:
+"ATI
+
+Manufacturer: huawei
+
+Model: EM770W
+
+Revision: 11.126.10.85.00
+
+IMEI: 357030027133509
+
++GCAP: +CGSM,+DS,+ES"
+
+Note: the detailed further AT command can refer to http://www.roman10.net/how-to-use-modem-at-commands-for-usb-modem-at-linux/
+   currently, only ATI command is OK
+
+| Expected Result |
+Some response can be received.
+
+|====================================================================
+
+<<<
-- 
1.8.0

